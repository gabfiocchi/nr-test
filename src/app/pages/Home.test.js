/* eslint-disable no-undef */
/* jest describe it */
import HomePage from './Home';

// eslint-disable-next-line compat/compat
const mockFetchPromise = Promise.resolve({
    // eslint-disable-next-line compat/compat
    json: () => Promise.resolve([{
        apps: [{
            version: 1,
            apdex: 100,
            name: 'Dummy'
        }]
    }])
});
window.fetch = jest.fn().mockImplementation(() => mockFetchPromise);

const home = new HomePage();
document.body.innerHTML = home.render();

describe('Home page', () => {
    it('should render the template', async () => {
        expect(home.render()).toMatchSnapshot();
    });
    it('should fecht the data', async () => {
        await home.fetchData();
        expect(home.state.hosts).toMatchObject([{
            apps: [{
                version: 1,
                apdex: 100,
                name: 'Dummy'
            }]
        }]);
    });
    it('should call to toggleView', () => {
        window.toggleView = jest.fn();
        document.toggleView();
        // expect(window.toggleView).toBeCalled();
        expect(home.state.viewGrid).toBeFalsy();
    });
});
