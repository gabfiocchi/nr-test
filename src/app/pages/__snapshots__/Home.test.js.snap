// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page should render the template 1`] = `
"
            <header>
                <h1>Apps by Host</h1>
                <h2>for user averylongemailadress@companyname.com</h2>
                <div class=\\"form-group\\">
                    <input type=\\"checkbox\\" id=\\"checkbox\\" checked onChange=\\"toggleView()\\">
                    <label for=\\"checkbox\\">Show as an awesome grid</label>
                </div>
            </header>
            <div class=\\"container\\">                
                
            </div>
        "
`;
